Automattic/jetpack
# My Jetpack: Activate product module after installing the standalone plugin

Fixes #30188.

## Proposed changes:

* Changed the installation method to call `Modules` `activate` function so the module for the product gets enabled as well
* Avoid activating the module for products that does not have the required plan to be activated
   * Search is the example, it requires a plan even for free usage (that also explain why the product stays inactive even when the plugin is installed and activated, and the action button stays as "Start for free")

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up a new JN site with only Jetpack
* Go to `Jetpack > My Jetpack` and open up the browser console
* Run the following command:

```
wp.data.dispatch( 'my-jetpack' ).installStandalonePluginForProduct( 'videopress' ).then(result => console.log(result))
```

* This will trigger the standalone plugin installation for VideoPress
* Notice that the VideoPress product card will enter a loading state
* After the loading finishes, the card will then change it's status to active, reflecting the correct status returned by the endpoint:

<img width="362" alt="Screen Shot 2023-04-20 at 15 26 42" src="https://user-images.githubusercontent.com/6760046/233455919-d8d33e5f-45ae-4d0b-b5a1-29cef3edbfa5.png">

<img width="538" alt="Screen Shot 2023-04-20 at 15 29 58" src="https://user-images.githubusercontent.com/6760046/233455948-667a0ab2-306d-44bf-9326-2da4ff46bbe5.png">

* You can test this for all the hybrid products: `videopress`, `social`, `backup` and `search`
* For some products, the next status after installing the plugin may not be `active` because they may require a plan to work
* For `Search`, for example, the status stays as `needs_purchase_or_free` resulting on a "Start for free" call-to-action button, but the plugin gets installed and activated correctly:

<img width="294" alt="Screen Shot 2023-04-20 at 15 33 26" src="https://user-images.githubusercontent.com/6760046/233456700-c5bd4bed-6f16-4270-969c-0fe1bc2d3194.png">

<img width="319" alt="Screen Shot 2023-04-20 at 15 34 06" src="https://user-images.githubusercontent.com/6760046/233456803-174e825e-02c6-456c-a24f-dc07fdd74e9e.png">

* The menu options under `Jetpack` also get enabled, as expected (you may need to refresh the page to see it):

<img width="173" alt="Screen Shot 2023-04-20 at 15 36 34" src="https://user-images.githubusercontent.com/6760046/233457535-f5f883cd-fa41-456c-a2c7-b280ef8003d1.png">
